### PR TITLE
chore: update required checks

### LINKED ISSUE #12: chore: update required checks
In https://github.com/googleapis/python-bigquery-magics/pull/7, I commented out several required checks. We should add these back in https://github.com/googleapis/python-bigquery-magics/blob/main/.github/sync-repo-settings.yaml

Some notes:

- Samples: blocked by https://github.com/googleapis/python-bigquery-magics/pull/9
- "Kokoro": maybe redundant with system tests and/or unit tests and linting. We may want to drop that in favor of the smaller test jobs.

### PR DESCRIPTION
**Pull request by chelsea-lin**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [X] Make sure to open an issue as an [issue](https://github.com/googleapis/python-bigquery-magics/issues/12) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [X] Ensure the tests and linter pass
- [X] Code coverage does not decrease (if any source code was changed)
- [X] Appropriate docs were updated (if necessary)

Fixes #12 🦕

### CODE REVIEW

File: .github/sync-repo-settings.yaml [hunk 1]
```diff
@@ -9,16 +9,17 @@ branchProtectionRules:
...
-    # - 'unit (3.8)'
-    # - 'unit (3.9)'
-    # - 'unit (3.10)'
-    # - 'unit (3.11)'
-    # - 'unit (3.12)'
-    # - 'cover'
-    # - 'Kokoro'
+    - 'unit (3.8)'
+    - 'unit (3.9)'
+    - 'unit (3.10)'
```
Comment (tswast) [inline .github/sync-repo-settings.yaml:17]: We aren't currently testing on 3.9 and 3.10. The hope is that testing on earlier and later Python versions should be sufficent, but if you think we need them, we'll need to also update `owlbot.py`.

```suggestion
    - 'unit (3.7)'
    - 'unit (3.8)'
```

File: .github/sync-repo-settings.yaml [hunk 2]
```diff
@@ -9,16 +9,17 @@ branchProtectionRules:
...
+    - 'unit (3.11)'
+    - 'unit (3.12)'
+    - 'cover'
+    - 'Kokoro'
```
Comment (tswast) [inline .github/sync-repo-settings.yaml:21]: This one is a duplicate with all our system tests. Let's add the specific system tests here instead.

File: noxfile.py [hunk 3]
```diff
@@ -98,15 +98,18 @@
 
 CURRENT_DIRECTORY = pathlib.Path(__file__).parent.absolute()
 
+# Excludes the duplicated tests:
```
Comment (tswast) [inline noxfile.py:101]: Owlbot is going to overwrite this. https://github.com/googleapis/synthtool/blob/master/synthtool/gcp/templates/python_library/noxfile.py.j2#L134-L143

We can do a replace rule if needed, but I don't think that's necessary if we remove the generic "kokoro" run.

File: .github/sync-repo-settings.yaml (hunk 2 above)
Comment (chelsea-lin) [reply .github/sync-repo-settings.yaml]: Addressed. Thanks!

File: .github/sync-repo-settings.yaml (hunk 1 above)
Comment (chelsea-lin) [reply .github/sync-repo-settings.yaml]: Corrected the version here. Thanks for catching that!

File: noxfile.py (hunk 3 above)
Comment (chelsea-lin) [reply noxfile.py]: Reversed the changes.

**Review by tswast — APPROVED** [review]

**Pull request merged**